terraform/main.tf

```diff
@@ -0,0 +1,550 @@
+}
+}
+
+resource "aws_scheduler_schedule" "media-sentiment-rss-pipeline-scheduler" {
+  name                         = "media-sentiment-rss-pipeline-scheduler"
+  schedule_expression_timezone = "Europe/London"
+  description                  = "Schedule to run ETL pipeline every hour"
+  state                        = "ENABLED"
+  
+  flexible_time_window {
+    mode = "OFF"
+  }
+
+  schedule_expression = "cron(0 * * * ? *)"
+
+  target {
+    arn      = "arn:aws:lambda:eu-west-2:129033205317:function:media-sentiment-rss-lambda"
+    role_arn = "arn:aws:iam::129033205317:role/media-sentiment-rss-scheduler-role"
+  }
+}
+
+# Media Sentiment RDS
+
+resource "aws_security_group" "media-sentiment-rds-sg" {
+  vpc_id = "vpc-0e0f897ec7ddc230d"
+  name = "media-sentiment-rds-sg"
+  ingress {
+    from_port = 5432
+    to_port = 5432
+    protocol = "tcp"
+    cidr_blocks = ["86.155.163.236/32"]
+  }
+  ingress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+}
+
+resource "aws_db_instance" "media-sentiment-rds" {
+  identifier="media-sentiment-rds"
+  allocated_storage    = 20
+  db_name              = var.database_name
+  engine               = "postgres"
+  engine_version       = "15"
+  instance_class       = "db.t3.micro"
+  username             = var.database_username
+  password             = var.database_password
+  parameter_group_name = "default.postgres15"
+  skip_final_snapshot  = true
+  performance_insights_enabled = false
+  db_subnet_group_name = "public_subnet_group"
+  publicly_accessible = true
+  vpc_security_group_ids = ["${aws_security_group.media-sentiment-rds-sg.id}"]
+}
+
+# ECS IAM Role
+
+
+resource "aws_iam_role" "ecs-task-execution-role" {
+  name = "ecs-task-execution-role"
+
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17",
+    Statement = [
+      {
+        Action = "sts:AssumeRole",
+        Principal = {
+          Service = "ecs-tasks.amazonaws.com"
+        },
+        Effect = "Allow",
+        Sid    = ""
+      }]
+  })
+}
+
+resource "aws_iam_role_policy_attachment" "ecs-task-execution-role-policy-attachment" {
+  role       = aws_iam_role.ecs-task-execution-role.name
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AmazonECSTaskExecutionRolePolicy"
+}
+
+# ECS Cluster
+
+resource "aws_ecs_cluster" "media-sentiment-cluster" {
+  name = "media-sentiment-cluster"
+}
+
+# Public Sentiment ECR
+
+resource "aws_ecr_repository" "media-sentiment-public-sentiment-ecr" {
+  name         = "media-sentiment-public-sentiment-ecr"
+  force_delete = true
+}
+
+# Public Sentiment ECS
+
+resource "aws_security_group" "media-sentiment-ecs-sg" {
+  name = "media-sentiment-ecs-sg"
+  vpc_id = data.aws_vpc.cohort-8-VPC.id
+    ingress {
+    from_port   = 443
+    to_port     = 443
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
```

Comment:
Why do we have inbound on port `443`? I'm not aware of us using `443` for accessing our resources?

Reply:
This is something ramiz had from a previous group, I am not too sure either. We can ask in next huddle.